src/test/java/org/jenkinsci/plugins/oic/OicSecurityRealmFipsTest.java

```diff
@@ -0,0 +1,98 @@
+package org.jenkinsci.plugins.oic;
+
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import java.io.IOException;
+import jenkins.security.FIPS140;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.reactor.ReactorException;
+import org.jvnet.hudson.test.FlagRule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.WithoutJenkins;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+
+public class OicSecurityRealmFipsTest {
+
+    @ClassRule
+    public static FlagRule<String> fipsFlag = FlagRule.systemProperty(FIPS140.class.getName() + ".COMPLIANCE", "true");
+
+    @Rule
+    public NonFailingOnStartupJenkinsRule j = new NonFailingOnStartupJenkinsRule();
+
+    @Test
+    @WithoutJenkins
+    public void settingNonCompliantValuesNotAllowedTest() throws IOException {
+        OicSecurityRealm realm = new OicSecurityRealm("clientId", Secret.fromString("secret"), null, false);
+        IllegalArgumentException ex = assertThrows(
+                IllegalArgumentException.class,
+                () -> new OicSecurityRealm("clientId", Secret.fromString("secret"), null, true));
+        assertThat(
+                "Exception contains the reason",
+                ex.getLocalizedMessage(),
```

Comment:
this will be unreliable if run with a different Locale?

Reply:
changed to getMessage